Util/geometry.py

```diff
@@ -12,15 +12,46 @@ def __init__(self, p1, p2):
         self.p1 = p1
         self.p2 = p2
 
+    @property
+    def normalize(self):
+        return normalize(self.p2 - self.p1)
```

Comment:
`normalize` seems to ba a strange function to have here, it's more a `direction`. The name direction implicitly say a vector from `p1` to `p2`.

Reply:
agreed